doc/cabal-commands.rst

```diff
@@ -273,7 +274,10 @@ cabal v2-exec
 
 ``cabal v2-exec [FLAGS] [--] COMMAND [--] [ARGS]`` runs the specified command
 using the project's environment. That is, passing the right flags to compiler
-invocations and bringing the project's executables into scope.
+invocations and bringing the project's executables into scope. 
+
+Note that unlike ``v2-run``, ``v2-exec`` will *not* (re)build anything; in particular,
+any changes since the last build will not be incorporated.
```

Comment:
There is conceptual difference that `v2-exec` executes *a command* in project environment, that command can (and often indeed is) completely unrelated to to the projects' packages/executables/... If that can be made clearer, then it should be more obvious *why* `v2-exec` doesn't rebuild anything.

In other words, I feel that comparing `v2-run` and `v2-exec` is on the level what they concretely do (e.g. rebuiding) is wrong, we should rather explain how they differ conceptually.

Also, I never needed `cabal v2-exec`, it's there for more or less legacy reasons. I'd be :+1: saying explicitly, "you are probably looking for `v2-run`" in the docs.

(Bad) examples of using `v2-exec` are
- `v2-exec which program` to find a built executable, there is now `cabal list-bin`
- `v2-exec doctest src/` which may or may not work, there isn't a right alternative at the moment though.

Sorry for not having concrete improvement, just a long comment.

Reply:
What about just adding

> If you want to run an executable from your project, you are probably looking for `v2-run`. `v2-exec` is more suited for running external programs that act on or inspect the project's environment.

at the end?

Still not perfect, but a net improvement over the status quo, no?

Reply:
and both?

> If you want to run an executable from your project, you are probably looking for v2-run. v2-exec is more suited for running external programs that act on or inspect the project's environment. Using it for running project executables can have unexpected behaviours like doesn't rebuild the executable.